homework/hw_w01.py

```diff
@@ -0,0 +1,210 @@
+                 user="postgres",
+                 host="localhost"):
+        """Constructor initialises connection and cursor"""
+        self.configure_conn(dbname, user, host)
+        self.connect()
+        self.open_cursor()
+
+    def __del__(self):
+        """Destructor to tidy up if required"""
+        if self.cur is not None:
+            self.close_cursor()
+        if self.conn is not None:
+            self.disconnect()
+
+    def connect(self):
+        """Connect to configured host"""
+        conn_str = f"dbname={self.dbname} user={self.user} host={self.host}"
+        self.conn = psycopg2.connect(conn_str)
+
+    def configure_conn(self, dbname, user, host):
+        """Configure connection"""
+        self.dbname=dbname
+        self.user=user
+        self.host=host
+
+    def disconnect(self, close_cursor_b=False):
+        """Close connection to database"""
+        if close_cursor_b:
+            self.close_cursor
+        self.conn.close()
+        self.conn = None
+
+    def is_connected(self):
+        """Check if connection is active"""
+        if self.conn is None:
+            return False
+        else:
+            return True
+
+    def open_cursor(self):
+        """Open database cursor"""
+        self.cur = self.conn.cursor()
+
+    def close_cursor(self):
+        """Close database cursor"""
+        self.cur.close()
+        self.cur = None
+
+    def select_query(self, query):
+        """Execute SELECT query"""
+        self.cur.execute(query)
+        return self.cur.fetchall()
+
+    def insert_query(self, query, var_tuple):
+        """Execute INSERT query"""
+        self.cur.execute(query, var_tuple)
+        self.conn.commit()
+
+    def _delete_query(self, query):
+        """Execute DELETE query :: Testing only"""
+        # Not a fan of providing direct delete methods, testing only...
+        self.cur.execute(query)
+        self.conn.commit()
+
+#-- #-- #-- #-- #-- #-- #-- #-- #-- #-- #-- #-- #-- #-- #-- #-- #-- #-- #--
+#
+# Module functions to construct queries and insert records
+#
+#-- #-- #-- #-- #-- #-- #-- #-- #-- #-- #-- #-- #-- #-- #-- #-- #-- #-- #--
+def build_query(row, table, query_type):
+    """Construct query from row for table"""
+    # Abstracted to allow for additon of UPDATE and SELECT later on
+    # if required
+    if query_type == "INSERT":
+        return build_insert_query(row, table)
+    else:
+        return ""
+
+def build_insert_query(row, table):
+    """Construct insert query from row for table"""
+    query_columns = ""
+    variable_list = list()
+    for key in row.keys():
+        # Insert is columns then values, so construct as two strings
+        query_columns += f"{key}, "
+        # Use format value to handle value types in SQL
+        variable_list.append(row[key])
+    values = "%s, " * len(row.keys())
+    # Index to -2 to remove ", " from last value
+    query = f"INSERT INTO {table} ({query_columns[:-2]}) VALUES ({values[:-2]})"
```

Comment:
This is what we talked about, the values should always be given SEPARATELY to the database driver (`psycopg2 `or it's higher-level abstractions such as `records` or `dataset`). 

Read the documentation of a library you are using! It tells you how to do it all!